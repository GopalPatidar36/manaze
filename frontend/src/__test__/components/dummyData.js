import {
  GET_ALL_TICKET,
  GET_CURRENT_USER,
  GET_TICKET,
  GET_CURRENT_USER_TICKET,
  DELETE_TICKET,
  UPDATE_USER,
  CREATE_TICKET,
  UPDATE_TICKET,
} from "../../Query/index";

export const login = {
  result: {
    data: {
      login: {
        token:
          "dummy.token.Ca1Ab6lXUn0v_nGvBnFNJDzd5SWr6hL-uEHukKW6IVeGA764MIH7pLsbrknBczubbfRedpgoOWeLp-e9jnOPFA3bjPoEYX8wTH60RUQtOiZv9ttBrxVXcpg4cRAuRCRsxyHp79riJyo8WBP_Jlwa9gnhj0vweazQeFwsq-cO1VS7EoCp_79vbr8draenM_Ly0F2ZDxxbyMguigqFYJN_lGO4V14aYhhVVZyqj2AVnp24NwdE258PH93fWscif2RE8Lxw344zoLx73S4bLo4YU9aq0Z7ZAwoATTKvC95_OIhhNFJ5QYx2HESuNYDa1Arfn4p0dqX1x9Q3ZQ1OsgMV0N7aSxRu09zwl_IL4B-3Grq2GMgFNW_icR46tSerV8wih2tZV6e18hFojtYu7I6LKWZlOYyTw2W3ZF3gZvCTjs__Gj0ZRYZJSpw1vTtxC0iR163XWDShYtWY6Rem94pSjmK_q923qe49GEMDE1F7NTC65e_kOQZwjMbu8AINIbx8uUlXei0LhdTCDXyQgUxpIjh8bTkysClVK5ZMk81w4zIGvd5h1t_4z3w2ts53iEfd_EiwJDTkxbY9GAsV5ijKJ-CQUbzPxuVuX_MTmi3kxarIdBU179EWA2cln9YMXYqWxHL2NLW0kHsoTlILzs89Mo7dvQ_Py-YyDZsHSGpNJIE",
        user: {
          uid: "236139b2-e7a5-49b8-9a56-f1b71cb12653",
          userEmail: "ramp@gmail.com",
          password: null,
          firstName: "ram",
          lastName: "Patidar",
          __typename: "RegistrationType",
        },
        __typename: "LoginTypes",
      },
    },
  },
};

export const getCurrentUSER = {
  request: { query: GET_CURRENT_USER },
  result: {
    data: {
      me: {
        __typename: "userFields",
        id: 11,
        uid: "236139b2-e7a5-49b8-9a56-f1b71cb12653",
        userEmail: "ramp@gmail.com",
        firstName: "ram",
        lastName: "Patidar",
        fullName: "ram Patidar",
        role: "member",
        createdAt: "2024-07-20T12:00:00Z",
        updatedAt: "2024-07-20T12:30:00Z",
      },
    },
  },
};

export const updateUser = {
  request: {
    query: UPDATE_USER,
    variables: { uid: "236139b2-e7a5-49b8-9a56-f1b71cb12653", firstName: "Gopal", lastName: "Patidar" },
  },
  result: {
    data: {
      updateUser: {
        __typename: "userFields",
        id: 11,
      },
    },
  },
};

export const createTicket = {
  request: {
    query: CREATE_TICKET,
    variables: {
      title: "dummpy title for test",
      description: "dummpy description for test",
      priority: "LOW",
      status: "INPROGRESS",
    },
  },
  result: { data: { createTicket: { id: 12 } } },
};

export const updateTicket = {
  request: {
    query: UPDATE_TICKET,
    variables: {
      id: 12,
      title: "update title for test",
      description: "update description for test",
      priority: "HIGH",
      status: "CLOSED",
    },
  },
  result: { data: { updateTicket: { id: 12 } } },
};

export const getUpdatedTicket = {
  request: {
    query: GET_TICKET,
    variables: { id: 12 },
  },
  result: {
    data: {
      ticketByID: {
        __typename: "Ticket",
        id: 12,
        title: "update title for test",
        description: "update description for test",
        priority: "HIGH",
        status: "CLOSED",
        createdAt: "234",
        updatedAt: "234",
        ticketUsersDetails: [
          {
            uid: "qalk-12321-w2dn",
            firstName: "gopal",
            lastName: "patidar",
            fullName: "gopal patidar",
          },
        ],
      },
    },
  },
};

export const getTicket = {
  request: {
    query: GET_TICKET,
    variables: { id: 12 },
  },
  result: {
    data: {
      ticketByID: {
        __typename: "Ticket",
        id: 12,
        title: "dummpy title for test",
        description: "dummpy description for test",
        priority: "LOW",
        status: "OPEN",
        createdAt: "234",
        updatedAt: "234",
        ticketUsersDetails: [
          {
            uid: "qalk-12321-w2dn",
            firstName: "gopal",
            lastName: "patidar",
            fullName: "gopal patidar",
          },
        ],
      },
    },
  },
};

export const deleteTicket = {
  request: {
    query: DELETE_TICKET,
    variables: { id: 12 },
  },
  result: { data: { deleteTicket: { id: 12 } } },
};

export const getAllTicket = {
  request: {
    query: GET_ALL_TICKET,
    variables: { offset: 0 },
  },
  result: {
    data: {
      ticketList: {
        __typename: "TicketList",
        count: 8,
        rows: [
          {
            __typename: "Ticket",
            id: 1,
            title: "create a ticket update  -61",
            description:
              "the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   ",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721052035000",
            createdAt: "1718715380000",
          },
          {
            __typename: "Ticket",
            id: 2,
            title:
              "ticket-702 testing is able to create a second number ticket  testing is able to create a second number ticket  testing is able to create a second number ticket testing is able to create a second number ticket  testing is able to create a second number ticket ",
            description: "testing is replace to create a second number ticket ",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721110760000",
            createdAt: "1718716225000",
          },
          {
            __typename: "Ticket",
            id: 5,
            title: "ticket-5-update",
            description: "1ticket-5-updated",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721040503000",
            createdAt: "1718716294000",
          },
          {
            __typename: "Ticket",
            id: 16,
            title: "ticket-20012",
            description: "ticket-20  update discription ",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721049603000",
            createdAt: "1718716667000",
          },
          {
            __typename: "Ticket",
            id: 17,
            title: "addTicket-17-12",
            description: "addTicket-17",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721052088000",
            createdAt: "1718716708000",
          },
          {
            __typename: "Ticket",
            id: 20,
            title: "title-20",
            description: "description-20",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721052088000",
            createdAt: "1718716708000",
          },
          {
            __typename: "Ticket",
            id: 21,
            title: "title-21",
            description: "description-21",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721052088000",
            createdAt: "1718716708000",
          },
          {
            __typename: "Ticket",
            id: 22,
            title: "title-22",
            description: "description-22",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721052088000",
            createdAt: "1718716708000",
          },
        ],
      },
    },
  },
};

export const getCurrentUSERTicket = {
  request: {
    query: GET_CURRENT_USER_TICKET,
    variables: { offset: 0 },
  },
  result: {
    data: {
      currentUserTicket: {
        __typename: "TicketList",
        count: 5,
        rows: [
          {
            __typename: "Ticket",
            id: 1,
            title: "current user a ticket update  -61",
            description:
              "the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   ",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721052035000",
            createdAt: "1718715380000",
          },
          {
            __typename: "Ticket",
            id: 2,
            title:
              "ticket-702 testing is able to create a second number ticket  testing is able to create a second number ticket  testing is able to create a second number ticket testing is able to create a second number ticket  testing is able to create a second number ticket ",
            description: "testing is replace to create a second number ticket ",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721110760000",
            createdAt: "1718716225000",
          },
          {
            __typename: "Ticket",
            id: 5,
            title: "ticket-5-update",
            description: "1ticket-5-updated",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721040503000",
            createdAt: "1718716294000",
          },
          {
            __typename: "Ticket",
            id: 16,
            title: "ticket-20012",
            description: "ticket-20  update discription ",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721049603000",
            createdAt: "1718716667000",
          },
          {
            __typename: "Ticket",
            id: 17,
            title: "addTicket-17-12",
            description: "addTicket-17",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721052088000",
            createdAt: "1718716708000",
          },
        ],
      },
    },
  },
};
