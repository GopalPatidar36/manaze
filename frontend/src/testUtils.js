// src/testUtils.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { Provider } from "react-redux";
import { MockedProvider } from "@apollo/client/testing";
import { BrowserRouter as Router } from "react-router-dom";
import store from "./store";

function render(ui, { preloadedState, apolloMocks, ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <MockedProvider mocks={apolloMocks} addTypename={false}>
          <Router>{children}</Router>
        </MockedProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
