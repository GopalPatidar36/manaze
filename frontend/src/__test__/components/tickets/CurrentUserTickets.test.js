import { screen, waitFor, fireEvent } from "@testing-library/react";
import CurrentUserTickets from "../../../components/tickets/CurrentUserTickets";
import { render } from "../../../testUtils";
import { getCurrentUSERTicket } from "../dummyData";

const mocks = [getCurrentUSERTicket];

describe("Test Current User Ticket", () => {
  let components;
  beforeEach(async () => {
    components = render(<CurrentUserTickets />, { apolloMocks: mocks });
  });

  test("Check current user ticket text", async () => {
    await waitFor(() => {
      expect(screen.getByText(/My Issue/)).toBeInTheDocument();
    });
  });

  test("Check Create text", async () => {
    await waitFor(() => {
      expect(screen.getByText(/Create Issue/)).toBeInTheDocument();
    });
  });
  test("Check ticket discription text", async () => {
    await waitFor(() => {
      const elements = screen.getAllByText(/ticket-20012/);
      expect(elements[0]).toBeInTheDocument();
    });
  });

  it("should match dom snapshot:-Backlog", async () => {
    await waitFor(() => {
      expect(components.asFragment()).toMatchSnapshot();
    });
  });
});
