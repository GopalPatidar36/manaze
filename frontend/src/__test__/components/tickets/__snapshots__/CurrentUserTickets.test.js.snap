// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Current User Ticket should match dom snapshot:-Backlog 1`] = `
<DocumentFragment>
  <div
    class="backlog"
  >
    <div
      class="backlog2"
    >
      <h3
        class=""
      >
        My Issue
      </h3>
      <button
        type="submit"
      >
        Create Issue
      </button>
    </div>
    <div
      class="DataTable"
    >
      <div
        style="display: flex; flex-wrap: wrap; margin-top: 5px;"
      >
        <input
          class="searchInput"
          placeholder="Search Ticket"
          value=""
        />
        <div
          style="flex-direction: column; display: flex; margin-left: 10px;"
        >
          <label
            class="label"
          >
            Priority
          </label>
          <select
            class="selectOption"
            id="country"
          >
            <option
              value=""
            >
              None
            </option>
            <option
              value="LOW"
            >
              Low
            </option>
            <option
              value="MEDIUM"
            >
              Medium
            </option>
            <option
              value="HIGH"
            >
              High
            </option>
          </select>
        </div>
        <div
          style="flex-direction: column; display: flex; margin-left: 10px;"
        >
          <label
            class="label"
          >
            Status
          </label>
          <select
            class="selectOption"
            id="country"
          >
            <option
              value=""
            >
              None
            </option>
            <option
              value="OPEN"
            >
              Open
            </option>
            <option
              value="INPROGRESS"
            >
              In Progress
            </option>
            <option
              value="CLOSED"
            >
              Close
            </option>
          </select>
        </div>
      </div>
      <div
        style="overflow-x: auto;"
      >
        <table>
          <tbody>
            <tr>
              <th
                class="tableHeader"
                id="title"
              >
                <span
                  style="cursor: pointer;"
                >
                  Title
                </span>
              </th>
              <th
                class="tableHeader"
                id="description"
              >
                <span
                  style="cursor: pointer;"
                >
                  Description
                </span>
              </th>
              <th
                class="tableHeader"
                id="priority"
              >
                <span
                  style="cursor: pointer;"
                >
                  Priority
                </span>
              </th>
              <th
                class="tableHeader"
                id="status"
              >
                <span
                  style="cursor: pointer;"
                >
                  Status
                </span>
              </th>
              <th
                class="tableHeader"
                id="createdAt"
              >
                <span
                  style="cursor: pointer;"
                >
                  CreatedAt
                </span>
              </th>
              <th
                class="tableHeader"
                id="updatedAt"
              >
                <span
                  style="cursor: pointer;"
                >
                  UpdatedAt
                </span>
              </th>
              <th
                id="actionHeader"
              />
            </tr>
          </tbody>
        </table>
      </div>
      <p
        style="align-self: center;"
      >
        Empty table Please create a ticket
      </p>
    </div>
  </div>
</DocumentFragment>
`;
